pkg/export/export.go

```diff
@@ -208,6 +208,38 @@ type ExporterOpts struct {
 	Efficiency EfficiencyOpts
 }
 
+func (opts *ExporterOpts) Default() {
+	if opts.Efficiency.BatchSize == 0 {
+		opts.Efficiency.BatchSize = BatchSizeMax
+	}
+	if opts.Efficiency.ShardCount == 0 {
+		opts.Efficiency.ShardCount = DefaultShardCount
+	}
+	if opts.Efficiency.ShardBufferSize == 0 {
+		opts.Efficiency.ShardBufferSize = DefaultShardBufferSize
+	}
+
+	if opts.Endpoint == "" {
+		opts.Endpoint = "monitoring.googleapis.com:443"
+	}
+	if opts.Compression == "" {
+		opts.Compression = CompressionNone
```

Comment:
Question: What compression does this control? Should we use gzip?

Reply:
This is an opt-in feature when users have many or large PodMonitorings and the corresponding ConfigMap reaches etcd limits of 1 MiB. The default is not changing here (the old code set the default when setting up the flag), and hopefully it's now clearer in the code what all of the defaults are.